docs/reference/commandline/run.md

```diff
@@ -697,30 +416,3 @@ if running on Windows server, any of these commands also result in `hyperv` isol
 $ docker run -d --isolation default busybox top
 $ docker run -d --isolation hyperv busybox top
 ```
-
-### Configure namespaced kernel parameters (sysctls) at runtime
-
-The `--sysctl` sets namespaced kernel parameters (sysctls) in the
-container. For example, to turn on IP forwarding in the containers
-network namespace, run this command:
-
-    $ docker run --sysctl net.ipv4.ip_forward=1 someimage
-
-
-> **Note**: Not all sysctls are namespaced. Docker does not support changing sysctls
-> inside of a container that also modify the host system. As the kernel
-> evolves we expect to see more sysctls become namespaced.
-
-#### Currently supported sysctls
-
-  `IPC Namespace`:
-
-  kernel.msgmax, kernel.msgmnb, kernel.msgmni, kernel.sem, kernel.shmall, kernel.shmmax, kernel.shmmni, kernel.shm_rmid_forced
-  Sysctls beginning with fs.mqueue.*
-
-  If you use the `--ipc=host` option these sysctls will not be allowed.
-
-  `Network Namespace`:
-      Sysctls beginning with net.*
-
-  If you use the `--network=host` option using these sysctls will not be allowed.
```

Comment:
**Exposing and publishing extra ports**

We should provide more information about `expose` here. As mentioned in
another comment, expose basically does _nothing_, it only adds metadata
on the container that;

- allows tooling to `docker inspect` the container and know what ports
  may be used (but no guarantees, because there may not be something
  listening)
- it is used for `-P` / `--publish-all` to automatically publish all
  ports that are marked as "exposed".

The important thing here is that;

- on the docker (container) network, _all_ ports of a container are "exposed",
  there are _no restrictions_ for other containers, or processes that
  have access to the network to connect to _any_ port of the container.
- if a container is using IPv6 and has a routable IP-address, all ports
  of the container are exposed, unless custom firewall/IPTables rules
  are present.


**Publishing ports**

In these examples;

- I suggest to remove the `--expose`, as it may add to the assumption
  that `--expose` is needed in order to publish a port (which is not
  the case), simply; `$ docker run -p 8080:80`

- Perhaps use a functional example image (`nginx:alpine` could be a 
  good choice as it's small, and actually serves something)
- All examples use `-it --rm`. This makes it difficult to inspect / try
  the example without opening a second shell. Consider using `-d`

Also, the `-p` / `--publish` flag has many options not mentioned here.
possibly some of them are mentioned in the networking section, but I
think we should have a central location to describe them. For example;

- `-p 80:80/tcp -p 80:80/udp` to publish udp/tcp
- mention that a single port can be mapped multiple times; `-p 81:80 -p 82:80`
- or on multiple interfaces `-p 192.168.10.20:80:80 -p 192.168.10.30:80:80`
- port ranges in which to pick a random target port `-p 8000-9000:80`
- a range of ports to publish on a range of random ports `-p 8000-8010`
- a range of ports to publish on a specified range of host ports: `-p 8000-8010:80-90`
- parts can be omitted when using these options, e.g.: `-p 192.168.10.20::80` to use a _random_ host port, but a _specific_ IP-address / interface

**Changing or disabling your container's network**

- I miss the tables :cry:. This may be highly personal, but having the
  tables allow me to quickly scan the page, and find which options are _available_.
  The table can be limited to a short description, and details following it.
  In the new format, there's a _lot_ of text, and it's hard to find a specific
  option. Maybe that's meant to be in "reference", but I think we should
  have this _somewhere_

- Using another container's network-space may deserve it's own section.
  we should add more information though (can be in a follow up), as it's
  an advanced feature, but very useful.

- Do we still have the information that was removed? For example,
  https://docs.docker.com/engine/userguide/networking/#/default-networks
  only mentions _"The host network adds a container on the hosts network
  stack. You’ll find the network configuration inside the container is
  identical to the host."_ The sections that were removed contained some
  details that should be available somewhere. No longer having a separate
  section for these network modes / networks makes it more difficult
  to refer users to the right information. For example, I can no longer
  link to https://docs.docker.com/engine/reference/run/#/network-host